src/directives/rcdni.js

```diff
@@ -0,0 +1,42 @@
+directives.directive('rcDni',['dniSemanticsValidationService','dniSyntaxValidationService',function(dniSemanticsValidationService,dniSyntaxValidationService){
+	return{
+		require: 'ngModel',					
+		link: function (scope, elm, attrs, ctrl){	
+			var validationToExecute = function (viewValue){
+				var currentTypedValue = viewValue;
+				validateDNI(currentTypedValue);
+			};
+
+			function validateDNI(dni){
+				if (validateDNISyntax(dni)){
```

Comment:
I don't like this too much. Let me explain: you see a function called validateDNI, where you expect the DNI to be validated, but however you can't see the function doing nothing.

I would make the validateSyntax and validateSemantics functions to return a boolean, and in this function I'd establish the validity of the DNI based on those booleans.

What do you think?